commcare_connect/opportunity/tasks.py

```diff
@@ -158,3 +159,22 @@ def _get_deliver_message(access: OpportunityAccess):
             "To maximise your payout complete all the required service delivery."
         ),
     )
+
+
+@celery_app.task()
+def send_payment_notification(opportunity_id: int, access_ids: list[int]):
+    opportunity = Opportunity.objects.get(pk=opportunity_id)
+    messages = []
+    for payment in Payment.objects.filter(opportunity_access__in=access_ids).select_related(
```

Comment:
This looks like it will send notifications for every payment the user has gotten in that opportunity, not just the new ones. I think you can fix this, by passing in the payment ids instead of the opportunity access ids. Then you will be able to get just the payments you want.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/221/commits/712e701faecd76b5f7898a20d6cae094b4f7fc22